src/canvas/PathIntersection.tsx

```diff
@@ -0,0 +1,78 @@
+import { Line } from 'canvas/Geometry/Line';
+import { PatternPath } from 'canvas/PatternPaths/PatternPath';
+import { BoundingBox } from './Geometry/BoundingBox';
+
+export class PathIntersection {
+    /* Checks for an intersection between one patternPath and an array of pattern paths */
```

Comment:
I think the comment could be more precise, since the method won't actually find an intersection anywhere on the path, just the last line segment. could be something like
Checks for an intersection between the segment formed by the last two points of the inputted path and an array of pattern paths

Reply:
That is a good point. I updated the comment.